src/units/unit.hpp

```diff
@@ -2014,6 +2014,17 @@ struct unit_movement_resetter
 	int moves_;
 };
 
+/**
+ * Optional parameter for get_checksum to use the algorithm of an older version of Wesnoth,
+ * thus preventing spurious OOS warnings while watching old replays.
+ *
+ * TODO: this needs a better name, and it ought to be in a namespace.
+ */
+enum class checksum_compatibility {
+	current_version,
+	version_1_16	/**< Included some of the flavortext from weapon specials. */
+};
```

Comment:
As the comment says, this needs a better name and should be in a namespace. Any suggestions?

Reply:
`checksum_version` maybe? Then the two values would be `current` and `1_16_or_older` (since all versions older than 1.17 didn't skip those attributes.

Reply:
Namespace could be just `compatibility` or `backwards_compatibility`.